packages/bindings/src/stargate/profiles/querier.rs

```diff
@@ -0,0 +1,619 @@
+        application: Option<&'b str>,
+        username: Option<&'b str>,
+        page_size: u64,
+    ) -> PageIterator<query_application_link_owners_response::ApplicationLinkOwnerDetails, Binary>
+    {
+        PageIterator::new(
+            Box::new(move |key, limit| {
+                self.query_application_link_owners(
+                    application.map(Into::into).clone(),
+                    username.map(Into::into).clone(),
+                    Some(PageRequest {
+                        key,
+                        limit: limit.into(),
+                        reverse: false,
+                        count_total: false,
+                        offset: None,
+                    }),
+                )
+                .map(|response| Page {
+                    items: response.owners,
+                    next_page_key: response.pagination.and_then(|response| {
+                        (!response.next_key.is_empty()).then_some(Binary::from(response.next_key))
+                    }),
+                })
+            }),
+            page_size,
+        )
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::mocks::mock_queriers::mock_desmos_dependencies;
+    use crate::profiles::mocks::{
+        MockProfilesQueries, MOCK_APPLICATION_LINK_APPLICATION, MOCK_APPLICATION_LINK_CLIENT_ID,
+        MOCK_APPLICATION_LINK_USERNAME, MOCK_CHAIN_LINK_ADDRESS, MOCK_CHAIN_LINK_CHAIN_NAME,
+        MOCK_USER,
+    };
+    #[test]
+    fn test_query_profile() {
+        let owned_deps = mock_desmos_dependencies();
+        let deps = owned_deps.as_ref();
+        let querier = ProfilesQuerier::new(&deps.querier);
+        let response = querier.query_profile(Addr::unchecked(MOCK_USER)).unwrap();
+        let expected = MockProfilesQueries::get_mocked_profile_response();
+        assert_eq!(expected, response)
+    }
+
+    #[test]
+    fn test_query_incoming_dtag_transfer_requests() {
+        let owned_deps = mock_desmos_dependencies();
+        let deps = owned_deps.as_ref();
+        let querier = ProfilesQuerier::new(&deps.querier);
+        let response = querier
+            .query_incoming_dtag_transfer_requests(Addr::unchecked(MOCK_USER), None)
+            .unwrap();
+        let expected = MockProfilesQueries::get_mocked_incoming_dtag_transfer_requests_response();
+        assert_eq!(expected, response)
+    }
+
+    #[test]
+    fn test_iterate_incoming_dtag_transfer_requests() {
+        let owned_deps = mock_desmos_dependencies();
+        let deps = owned_deps.as_ref();
+        let querier = ProfilesQuerier::new(&deps.querier);
+        let mut it =
+            querier.iterate_incoming_dtag_transfer_requests(Addr::unchecked(MOCK_USER), 10);
+        let expected = MockProfilesQueries::get_mocked_incoming_dtag_transfer_requests_response();
+        assert_eq!(expected.requests[0], it.next().unwrap().unwrap(),);
+        assert!(it.next().is_none());
+    }
+
+    #[test]
+    fn test_query_chain_links() {
```

Comment:
Any types serialization will lost type_url field since serde_json_wasm doesn't implement Map. So chain links related tests are always failed.

Reply:
Waiting cosmwasm v1.2.0 to fix the issue.

Reply:
Backporting v1.2.0 to enable the Any structure support.